docs/source/DeveloperGuide/OAuth2Authorization/index.rst

```diff
@@ -0,0 +1,194 @@
+.. _oauth2_authorization:
+
+####################
+OAuth2 authorization
+####################
+
+`OAuth2 <https://oauth.net/2/>`_ authorization is being implemented in DIRAC for transitioning to using access tokens to manage jobs and data, and for simplifying the authorization process for users.
+
+.. contents::
+
+*************
+The main goal
+*************
+
+The main goal is to expand the capabilities of DIRAC in terms of interoperability with third-party systems that support OAuth 2 authorization.
+Enabling users to access DIRAC resources not only through a proxy certificate, but also with access tokens obtained through Identity Providers.
+
+*******************
+OAuth 2.0 framework
+*******************
+
+The OAuth 2.0 authorization framework is a protocol that allows a user to grant a third-party web site or application access to the user's protected resources, without necessarily revealing their long-term credentials or even their identity.
+There are already many articles to familiarize yourself with this framefork, for example `Auth0 Docs <https://auth0.com/docs/authorization/protocols/protocol-oauth2>`_ or `RFCs <https://oauth.net/>`_
+
+.. image:: /_static/Systems/FS/OAuth2/OAuth2Roles.png
+   :alt: OAuth 2.0 roles in DIRAC.
+
+OAuth 2.0 roles
+===============
+
+An OAuth 2.0 flow has the following `roles <https://datatracker.ietf.org/doc/html/rfc6749#section-1.1>`_:
+
+ - **Resource Owner** - Entity that can grant access to a protected resource. In the context of DIRAC, these are DIRAC users.
+ - **Resource Server** - Server hosting the protected resources. In the context of DIRAC, this is DIRAC backend components like a DIRAC services.
+ - **Client** - Application requesting access to a protected resource on behalf of the *Resource Owner*. In the context of DIRAC, these are DIRAC client installations.
+ - **Authorization Server** - Server that authenticates the *Resource Owner* and issues access tokens after getting proper authorization. In the context of DIRAC, this is DIRAC Authorization Server.
+
```

Comment:
In this schema, where do DIRAC agents fit?

We have DIRAC agents that authenticate themselves using the server certificate of the machine where they are running, while some use a proxy of a user (from the ProxyManager) to authenticate to external services.

Reply:
When an **Agent** (or **Service**) makes a request to another **Service**, in the _OAuth2_ paradigm, the **Agent** acts as a **Client**, a **Service** accessed by the **Agent** - a **resource server** or a **protected resource**.

If the **Agent** uses a _host certificate_, this is a type of access similar to accessing with _client credentials_ (when _client_id_ and _client_secret_ are passed in requests) and is a request for a **protected resource** on behalf of the **client**. If an **agent** uses a _proxy_ (or user _access token_), it is a request for a **protected resource** on behalf of the **resource owner**.

It looks like I need to add one more diagram showing the interaction between DIRAC components (without direct user involvement).

Reply:
> It looks like I need to add one more diagram showing the interaction between DIRAC components (without direct user involvement).

Thanks! that would help!

Reply:
added